shap/explainers/_tree.py

```diff
@@ -182,7 +183,7 @@ def __init__(
             self.data = data
         if self.data is None:
             feature_perturbation = "tree_path_dependent"
-            #warnings.warn("Setting feature_perturbation = \"tree_path_dependent\" because no background data was given.")
+            logging.warn("Setting feature_perturbation = \"tree_path_dependent\" because no background data was given.")
```

Comment:
```suggestion
            logging.warning("Setting feature_perturbation = \"tree_path_dependent\" because no background data was given.")
```

I believe `logging.warn` is deprecated in favour of `logging.warning`.

Even better, the message could come from a module-level logger rather than the root logger: `log = logging.getLogger(__name__)`.